### PR TITLE
Clientservice gRPC scaffolding + clang-tidy changes

### PR DESCRIPTION
**Pull request by robem**
This change introduces a simple gRPC server that does nothing. It is basic scaffolding in order to have cleaner PRs later. The server is working and I tested its endpoints with`grpc_cli`.

`make tidy-check` wasn't happy with this PR. The investigation showed that it would be best to have a proper way of excluding files from clang-tidy. Therefore, part of this change introduces `.clang-tidy-ignore` which takes a glob file name pattern to be ignored. Please check the commits for details.

### CODE REVIEW

File: client/clientservice/src/event_service.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+using grpc::Status;
+using grpc::ServerContext;
+using grpc::ServerWriter;
+
+using vmware::concord::client::v1::StreamEventGroupsRequest;
+using vmware::concord::client::v1::EventGroup;
+
+namespace concord::client {
+
+Status EventServiceImpl::StreamEventGroups(ServerContext* context,
```
Comment (f-squirrel) [inline client/clientservice/src/event_service.cpp:25]: I know that this is just the beginning but would you mind changing the method's name to `camelCase`?
I think that having `camelCase` is the only thing we've managed to reach consensus for :)

File: client/clientservice/src/event_service.cpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                                           const StreamEventGroupsRequest* request,
+                                           ServerWriter<EventGroup>* stream) {
+  std::cout << "EventServiceImpl::StreamEventGroups called" << std::endl;
```
Comment (f-squirrel) [inline client/clientservice/src/event_service.cpp:28]: Do we plan to add `log4cplus` logger for this class in the future?

File: client/clientservice/src/event_service.cpp (hunk 1 above)
Comment (robem) [reply client/clientservice/src/event_service.cpp:25]: I cannot. This is the protobuf/gRPC generated interface.

File: client/clientservice/src/event_service.cpp (hunk 2 above)
Comment (robem) [reply client/clientservice/src/event_service.cpp:28]: Yes, absolutely, I just wanted to get some basic scaffolding going and add proper logging and configuration later. Also, the whole Concord Client library is still missing. Lots to do!

File: client/clientservice/src/event_service.cpp (hunk 1 above)
Comment (f-squirrel) [reply client/clientservice/src/event_service.cpp:25]: Oh, I did not realize that it was part of the interface, my bad!

**Review by f-squirrel — APPROVED** [review]
LGTM!

**Comment by shgandhi** [comment]
Thanks for the change @robem, this looks great! 💪

File: Makefile [hunk 3]
```diff
@@ -15,6 +15,7 @@ CONCORD_BFT_CONTAINER_CC?=clang
 CONCORD_BFT_CONTAINER_CXX?=clang++
 CONCORD_BFT_CMAKE_BUILD_TYPE?=Release
 CONCORD_BFT_CMAKE_BUILD_TESTING?=TRUE
+CONCORD_BFT_CLANG_TIDY?=/concord-bft/tools/run-clang-tidy.py
```
Comment (f-squirrel) [inline Makefile:18]: Could you please build the path using the top-level variable `CONCORD_BFT_TARGET_SOURCE_PATH`?

File: Makefile [hunk 4]
```diff
@@ -170,7 +171,8 @@ tidy-check: gen_cmake ## Run clang-tidy
 		make -C ${CONCORD_BFT_RECONFIGURATION_CMF_PATHS} &> /dev/null && \
 		make -C ${CONCORD_BFT_CLIENT_PROTO_PATH} &> /dev/null && \
 		make -C ${CONCORD_BFT_THIN_REPLICA_PROTO_PATH} &> /dev/null && \
-		run-clang-tidy-10 2>&1 | tee clang-tidy-report.txt | ( ! grep 'error:\|note:' ) && \
+		cp ../.clang-tidy-ignore . && \
```
Comment (f-squirrel) [inline Makefile:174]: Nit: We may add a parameter instead of copying the file to the build directory.

File: .clang-tidy [hunk 5]
```diff
@@ -32,7 +32,9 @@ WarningsAsErrors:   'bugprone-*,
 # -misc-unused-parameters - too many errors, fix later
 
 # Include all headers in the project
-HeaderFilterRegex: 'concord-bft'
+# Except for protobuf/gRPC generated header files
+# Note: '(?<!' starts a negative look behind expression
```
Comment (f-squirrel) [inline .clang-tidy:36]: Good comment!

**Review by f-squirrel — APPROVED** [review]
Looks great, @robem !
Thank you for adding it!

File: .clang-tidy (hunk 5 above)
Comment (robem) [reply .clang-tidy:36]: I took me quite some time to figure this out so I thought it is worth a comment.

**Pull request merged**